### DISCUSSION TITLE
Płatnik w Azure SQL

### DISCUSSION THREAD

**Discussion by adamhatak** [AzureCommunityPL/group#26] [Pomoc] [answered]
Cześć,
chciałbym zapytać czy ktoś ma doświadczenia w migracji bazy Płatnika (ten od ZUS) do Azure SQL? Czy to w ogóle zadziała? Plan jest następujący:
- baza SQL na Azure SQL
- aplikacja Płatnika zainstalowana i udostępniona na WVD

Z góry dzięki za podpowiedzi, komentarze itp...

**Comment by kamil-mrzyglod** [comment]
Pamiętam, że jakiś czas temu to analizowałem i masz w Płatniku normalnie możliwość archiwizacji bazy danych i wskazania nowej. Płatnik normalnie wspiera SQL więc nie ma znaczenia czy to SQL Server czy Azure SQL. Cały proces jest chyba nawet w dokumentacji Płatnika opisany.

**Reply by adamhatak** [reply]
Kamil, 
Dzięki za odpowiedź. Spróbuję, wykonam kopię, zmigruję bazę do Azure i przełączę Płatnika na nową bazę danych w Azure.
Podzielę się informacjami co i jak.

**Reply by smereczynski** [reply]
"nie ma znaczenia czy to SQL Server czy Azure SQL" to chyba zbyt daleko idące stwierdzenie :D Wierz mi, że wielu aplikacjom robi to kolosalną różnicę, choćby dlatego, że w Azure SQL nie użyjesz USE do zmiany kontekstu bazy, a wiele z tych softów leci na kilku bazach.

**Reply by adamhatak** [reply]
Michał, 
oczywiście, że jest różnica pomiędzy Azure SQL Single Database, a SQL Server. Już nawet na samym T-SQLu występują różnice (przynajmniej jakiś czas temu tak było ;)). I tych różnic trochę się obawiam, że mogą coś wywalić w Płatniku. Chyba nie pozostaje nic innego jak przetestować to: kopia, migracja do Azure SQL, przepięcie Płatnika i testy.

**Reply by kamil-mrzyglod** [reply]
@smereczynski No tutaj masz rację - gdzieś na poziomie niektórych kwerend czy opcji samego serwera coś się rozjechać może. Skrót myślowy - chodziło mi, że jak zarządzalny Azure SQL nie zabangla to zawsze są inne opcje jeszcze na SQL w Azure, niekoniecznie managed a skoro Płatnik SQLa wspiera, to któraś ma szansę zadziałać :)

**Reply by TomaszOledzki** [reply]
Azure SQL Database != SQL Server - to już wiemy, a tu link: https://docs.microsoft.com/en-us/azure/azure-sql/database/transact-sql-tsql-differences-sql-server

Nie widzę w specyfikacji Płatnika, jakoby Azure SQL Database był wspierany: https://www.zus.pl/firmy/program-platnik/o-programie-platnik/wymagania-sprzetowe-wersja-10.02.002.

Pisałeś w ogóle do vendora czy Płatnik działa na Azure SQL Database? ;) Czy testowali? Czy wspierają takie rozwiązanie? W sumie to tylko dane ZUS, co może się stać?

Obczaj to:
https://docs.microsoft.com/en-us/azure/azure-sql/database/migrate-to-database-from-sql-server#migrate-to-a-single-database-or-a-pooled-database

**Reply by kamil-mrzyglod** [reply]
@TomaszOledzki - pytanie czy w ogóle oni są świadomi istnienia Azure SQL ;) Tutaj pytanie czy po prostu nie szybciej wyeksportować dane z Płatnika, postawić Azure SQL, podpiać Płatnika i zobaczyć czy działa bez szukania, pytania itd.

**Reply by smereczynski** [reply]
Pewnie, że szybciej sprawdzić. Tylko nie masz testów jednostkowych, żeby sprawdzić czy aplikacja nie jebnie w jakimś scenariuszu :)

**Reply by TomaszOledzki** [reply]
Ano :).

No to tak jak pisałem - co może się stać. To "tylko" dane ZUS ;). Kwestia jak dobre testy jest w stanie przeprowadzić @adamhatak i czy chce potem z tym iść na PROD mając w świadomości, że i tak ma rozwiązanie niewspierane przez vendora.

**Reply by kamil-mrzyglod** [reply]
To zawsze zostaje postawienie VM wraz z oficjalnie wspieraną wersją SQL. Tak teraz spojrzałem na wymagania i widzę, że ktoś popłynął z tłumaczeniem:
![image](https://user-images.githubusercontent.com/11725731/91068445-97577300-e634-11ea-8305-a2b36ea83b0b.png)

**Reply by MichalJankowskii** [reply]
Hej,

Mi taki manewr (migracja na Azure SQL) się nie udała, ale robiłem to dawno temu. Może coś się pozmieniało. Wtedy brakowało jakiś funkcji po stronie Azure SQL. Tak jak koledzy pisali Azure SQL <> MS SQL.

Patrząc na wspierane wersje jest duże ryzyko, że nie zadziała automatycznie.

Natomiast część rzeczy można samodzielnie naprawić - przykładowo dodając brakujące funkcje. Choć po pierwsze nie wszystko się da.

Pytanie czy jest to gra warta świeczki?

**Comment by adamhatak** [comment]
@TomaszOledzki @smereczynski  Niestety nie jestem w stanie przeprowadzić dobrych testów :) Faktycznie chyba lepiej odpuścić i zrobić na VMz SQLem i wykorzystać tę instancję też do innych rzeczy. 

Dzięki za udział w dyskusji.

**Reply by mgrabarz** [reply]
Dla mnie pytanie od @TomaszOledzki jest dalej aktualne. Czy pytałeś INSERTa, czy wspierają Azure SQL w swoim produkcie. Jeśli tak, pytasz o metodę migracji, jesli nie - wypytujesz klienta,czy są ok z rozwiązaniem niewspieranym przez dostawcę i dopiero wtedy zaczynasz swoją zabawę.

Ja przykładowo bym zaczął od zrobienia bacpaca i próby importu. Przy okazji, jak pisał @MichalJankowskii  poprawiając pojawiające się zgrzyty.

**Reply by adamhatak** [reply]
@mgrabarz Nie kontaktowałem się z Asseco (autor Płatnika) w kwestii wsparcia dla Azure. Na oficjalnej stronie Płatnika nie widzę, aby było zaznaczone wsparcie dla Azure. Jest wsparcie dla MS SQL Express (tak jak @TomaszOledzki wrzucił link https://www.zus.pl/firmy/program-platnik/o-programie-platnik/wymagania-sprzetowe-wersja-10.02.002), ale to już wiemy, że instalacja SQL Server (nawet Express) to nie to samo do Azure SQL Single database. 

Liczyłem, że może ktoś mierzył się z tym tematem i może podzielić się doświadczeniami. Temat może być ciekawy, jeżeli dodamy do tego Windows Virtual Desktop i udostępnienie tam Płatnika z bazą Azure SQL. 

Jeżeli uda mi się coś zrobić w tym temacie, podzielę się tutaj krótkim opisem działań.

**Reply by TomaszOledzki** [reply]
@adamhatak jeśli ta instancja może być też do innych rzeczy, to rozważ Azure SQL Managed Instances. To też jest PaaS, a jest zgodne z SQL Server. Być może po wrzuceniu tych innych rzeczy, koszt SQL MI nie będzie tak straszny w porównaniu do SQL na VM - w końcu odpada Ci koszt związany z administracją Windowsem i VMką.

https://docs.microsoft.com/en-us/azure/azure-sql/managed-instance/sql-managed-instance-paas-overview

**Comment by MariuszFerdyn** [comment, marked as answer]
Azure SQL nie jest wspiarany dla Płatnika. Nawet jakby się to udało przenieść to nie polecam, bo przy aktulizacjach może być ciężko.
Managed Instance powinno działać, ale oficjalnie nie będzie wspierane.

**Reply by adamhatak** [reply]
Proponuję, aby odpowiedź @MariuszFerdyn była podsumowaniem tej dyskusji. Nie chcesz problemów z działaniem aplikacji klasycznych wykorzystujących SQLa, nie pchaj się w Azure SQL. Nawet za cenę niskiej opłaty za Single Database.
